editor.planx.uk/src/@planx/components/NumberInput/model.ts

```diff
@@ -52,6 +54,19 @@ export const numberInputValidationSchema = (input: NumberInput) =>
         }
         return value === "0" ? true : Boolean(parseNumber(value));
       },
+    })
+    .test({
+      name: "check for a whole number",
+      message: "Enter a whole number",
+      test: (value: string | undefined) => {
+        if (!value) {
```

Comment:
Added this to cover the `string | undefined' type without mudding conditional check on the `Number.isInteger`